### PR TITLE
Fix:Service Account Annotation Formatting Error

### PR DESCRIPTION
**Pull request by 7h3-3mp7y-m4n**
**Please provide a description of this PR:**
This pr held to fix the [issue](https://github.com/istio/istio/issues/51289) by removing the dashes

**To help us figure out who should review this PR, please put an X in all the areas that this PR affects.**

- [ ] Ambient
- [ ] Configuration Infrastructure
- [ ] Docs
- [ ] Dual Stack
- [x] Installation
- [ ] Networking
- [ ] Performance and Scalability
- [ ] Policies and Telemetry
- [ ] Security
- [] Test and Release
- [ ] User Experience
- [ ] Developer Infrastructure

### CODE REVIEW

**Review by hanxiaop — COMMENTED** [review]
/ok-to-test

File: manifests/charts/istiod-remote/templates/serviceaccount.yaml [hunk 1]
```diff
@@ -13,7 +13,7 @@ metadata:
   labels:
     app: istiod
     release: {{ .Release.Name }}
-  {{- if .Values.pilot.serviceAccountAnnotations -}}
+  {{- if .Values.pilot.serviceAccountAnnotations }}
```
Comment (hanxiaop) [inline manifests/charts/istiod-remote/templates/serviceaccount.yaml:16]: The istiod chart is also in the scope, please update https://github.com/istio/istio/tree/master/manifests/charts/istio-control/istio-discovery and run `make gen`

Comment (7h3-3mp7y-m4n) [reply manifests/charts/istiod-remote/templates/serviceaccount.yaml:16]: okay

**Comment by dgn** [comment]
Please also add a release note (check for an example in the `releasenotes` directory). You can link the fixed issue there as well

File: manifests/charts/istiod-remote/templates/serviceaccount.yaml (hunk 1 above)
Comment (7h3-3mp7y-m4n) [reply manifests/charts/istiod-remote/templates/serviceaccount.yaml:16]: @zirain  can you be more brief ?, I'm just a beginner

Comment (zirain) [reply manifests/charts/istiod-remote/templates/serviceaccount.yaml:16]: seems there's no such tests, let's do it in an sperated PR

**Review by zirain — APPROVED** [review]

File: manifests/charts/istiod-remote/templates/serviceaccount.yaml (hunk 1 above)
Comment (hanxiaop) [reply manifests/charts/istiod-remote/templates/serviceaccount.yaml:16]: @zirain gen test will fail in this case

**Pull request merged**